settings/factory.py

```diff
@@ -69,6 +71,8 @@ def get_reader(self):
 
     if reader_conf["name"] == 'kafka_reader':
       print("creating Kafka reader..")
+      reader_conf["connection"]["settings"]["group.id"] = self.get_consumer_group_id(reader_conf)
```

Comment:
this should also support the default method of picking consumer group directly from the app-config.json

Reply:
- yes, if someone wants default method they must not add that mode field